apt/minimization/orderedfeatureminimizer.py

```diff
@@ -0,0 +1,458 @@
+            else:
+                untouched_features.append(feature_name)
+
+        return {
+            "ranges": numerical_thresholds,
+            "categories": categories_per_feature,
+            "untouched": untouched_features,
+        }
+
+    @classmethod
+    def _transform_numerical_feature(cls, dt: Tree, X: np.ndarray, medians: list, depth: int, node_id: int = 0):
+        if X.size == 0:
+            return
+        # check if each split results in left and right children
+        threshold = dt.threshold[node_id]
+        left_id = dt.children_left[node_id]
+        right_id = dt.children_right[node_id]
+        if (left_id < 0 or right_id < 0) or depth == 0 or medians[left_id] is None or medians[right_id] is None:
+            return medians[node_id]
+        if left_id >= 0:
+            X[X <= threshold] = cls._transform_numerical_feature(dt, X[X <= threshold], medians, depth - 1, left_id)
+        if right_id >= 0:
+            X[X > threshold] = cls._transform_numerical_feature(dt, X[X > threshold], medians, depth - 1, right_id)
+        return X
+
+    @classmethod
+    def _transform_categorical_feature(cls, dt: Tree, X: np.ndarray, majors: list, depth: int, node_id: int = 0):
+        # TODO Debug and make sure it works
+        if X.size == 0:
+            return
+        split_feature = dt.feature[node_id]
+        threshold = dt.threshold[node_id]
+        left_id = dt.children_left[node_id]
+        right_id = dt.children_right[node_id]
+        if (left_id < 0 or right_id < 0) or depth == 0 or majors[left_id] is None or majors[right_id] is None:
+            X[:] = 0
+            X[:, majors[node_id]] = 1
+            return X
+        y = X[:, split_feature]
+        if left_id >= 0:
+            X[y <= threshold] = cls._transform_categorical_feature(dt, X[y <= threshold], majors, depth - 1,
+                                                                   left_id)
+        if right_id >= 0:
+            X[y > threshold] = cls._transform_categorical_feature(dt, X[y > threshold], majors, depth - 1,
+                                                                  right_id)
+        return X
+
+    @classmethod
+    def _transform_in_place(cls, dts, X, depths: Dict[str, int], numerical_features, categorical_features,
+                            feature_indices, generalizations_arrays):
+        for feature in numerical_features:
+            index = feature_indices[feature][0]
+            X[:, index] = cls._transform_numerical_feature(dts[feature].tree_, X[:, index],
+                                                           generalizations_arrays[feature], depths[feature], 0)
+
+        for feature in categorical_features:
+            indices = feature_indices[feature]
+            X[:, indices] = cls._transform_categorical_feature(dts[feature].tree_, X[:, indices],
+                                                               generalizations_arrays[feature], depths[feature], 0)
+
+    @classmethod
```

Comment:
I think we agreed there is no need for class methods. These can be changed to static or removed from the class to some utils file.